### ISSUE TITLE
security/acme-client: logging is broken

### ISSUE THREAD

**Issue by L1ghtn1ng** [opnsense/plugins#2555]
This is due to phalcon4 as far as I can tell, as when you look under system log tab it just says loading and never loads, under the acme log tab I have entries from the 1-3-2021 but nothing else  cc @fraenki

**Comment by fraenki** [comment]
Version? Errors or warnings in `System: Log Files: Backend` or `System: Log Files: General`? Does `/tmp/PHP_errors.log` exist?

**Comment by L1ghtn1ng** [comment]
Using OPNsense 21.7.3_3 so the latest plugin version, only noticed this due to the whole letencrypt root cert expiring
________________________________

**Comment by fraenki** [comment]
I can't reproduce this issue, hence I was asking for logs and details. :)

**Comment by L1ghtn1ng** [comment]
Can give the logs after work with screen shots
________________________________

**Comment by L1ghtn1ng** [comment]
![image](https://user-images.githubusercontent.com/3628354/135775424-9bdad5cd-93c1-4780-a1de-b5959992c332.png)
This happens when I click to go on to the 2nd page of the logs on the system log tab

From also looking at the logs in a root shell the acme log tab in the above screenshot is reading data from /var/log/acmeclient.log which matches what I see in the log file for it but it is actually logging to /var/log/acme.sh.log so it it needs to be changed to that to read from. @fraenki  sorry for the delay in getting you this information

**Comment by razamatan** [comment]
i'm sorta hitting this issue as well.  unfortunately acmeclient doesn't appear to even start up, and i have no logs being output at all..

**Comment by L1ghtn1ng** [comment]
I have changed sensitive stuff to example or example.com, this is the output as follows;

```AcmeClient: running acme.sh command: /usr/local/sbin/acme.sh --issue --syslog 7 --debug --server 'letsencrypt' --dns 'dns_cf' --dnssleep '30' --home '/var/etc/acme-client/home' --certpath '/var/etc/acme-client/certs/example/cert.pem' --keypath '/var/etc/acme-client/keys/example/private.key' --capath '/var/etc/acme-client/certs/example/chain.pem' --fullchainpath '/var/etc/acme-client/certs/example/fullchain.pem' --domain 'example.com' --days '1' --force --keylength '4096' --accountconf '/var/etc/acme-client/accounts/example_prod/account.conf'```

**Comment by L1ghtn1ng** [comment]
@fraenki you managed to make any progress on this?

**Comment by fraenki** [comment]
@L1ghtn1ng nope, I have no idea unfortunately.

**Comment by kulikov-a** [comment]
hi all
may be its syslog/syslog-ng switch issue again (like as HAProxy)?
if i add acmeclient.conf file with content
```
###################################################################
# Local syslog-ng configuration filter definition [acmeclient].
###################################################################
filter f_local_acmeclient {
    program("acme.sh");
};

```
to the `opnsense/service/templates/OPNsense/Syslog/local/` dir and reload syslog-ng, logging starts working

**Comment by L1ghtn1ng** [comment]
Nice, can you create a PR for it please?


On 14 Nov 2021, at 07:01, kulikov-a ***@***.***> wrote:

﻿

hi all
may be its syslog/syslog-ng switch issue again (like as HAProxy)?
if i add acmeclient.conf file with content

###################################################################
# Local syslog-ng configuration filter definition [acmeclient].
###################################################################
filter f_local_acmeclient {
    program("acme.sh");
};



to the opnsense/service/templates/OPNsense/Syslog/local/ dir and reload syslog-ng, logging starts working

**Comment by kulikov-a** [comment]
@L1ghtn1ng 
can you test it please?
`opnsense-patch -a kulikov-a -c plugins 0ddbdd5`

**Comment by L1ghtn1ng** [comment]
not seeing anything in the acme log tab still

**Comment by L1ghtn1ng** [comment]
The system tab is fine as that has always worked

**Comment by L1ghtn1ng** [comment]
@fraenki in this file https://github.com/opnsense/plugins/blob/master/security/acme-client/src/opnsense/mvc/app/views/OPNsense/AcmeClient/logs.volt L71 and L58 are duplicated is that needed?

**Comment by kulikov-a** [comment]
> L71 and L58 are duplicated is that needed?

they are different. the first for the system log, the second for acme

> not seeing anything in the acme log tab still

sorry, clog disabled? syslog-ng reloaded? cert renew forced?

**Comment by L1ghtn1ng** [comment]
@kulikov-a syslog-ng has been restarted and new cert forced, clog is disabled and thanks for letting me know that they are different

**Comment by kulikov-a** [comment]
> syslog-ng has been restarted

ah. sorry again (i tested it without acme templates targets. placed acmeclient.conf dircetly in syslog-ng templates dir). patch worked for me after opnsense reboot only

**Comment by L1ghtn1ng** [comment]
@kulikov-a After the reboot and doing another force cert renewal it now works !!!! thanks for this much appreciated. A PR incoming soon I see in the future :-)

**Comment by L1ghtn1ng** [comment]
unless @fraenki pulls it into his PR that he has currently open?

**Comment by kulikov-a** [comment]
yes, lets wait for @fraenki ) 
@fraenki if you agree with the proposed solution, i'm ready to do whatever is more convenient for you: make a pr or you just squash it to your release while it is under review? thanks!

**Comment by fraenki** [comment]
@kulikov-a Thanks for stepping in, much appreciated! :) I've cherry-picked your commit for the next release:
https://github.com/opnsense/plugins/pull/2647/commits

This will be available in os-acme-client 3.5.

**Comment by kulikov-a** [comment]
thanks a lot )
